source/math/math_function.py

```diff
@@ -253,15 +253,15 @@ def slant_asymptotes(self, symbol: sy.Symbol) -> set:
         # TODO periodic function
 
         k = sy.limit(self.simplified_expr / symbol, symbol, sy.oo)
-        if k.is_finite:
+        if k.is_number:
             b = sy.limit(self.simplified_expr - k * symbol, symbol, sy.oo)
-            if b.is_finite:
+            if b.is_number:
                 ans.add(k * symbol + b)
 
         k = sy.limit(self.simplified_expr / symbol, symbol, -sy.oo)
-        if k.is_finite:
+        if k.is_number:
             b = sy.limit(self.simplified_expr - k * symbol, symbol, -sy.oo)
-            if b.is_finite:
+            if b.is_number:
```

Comment:
We must add "**and k.is_finite**". Github doesn't allow to make a suggestion on deleted lines, so it cannot be done automatically